src/System.Management.Automation/namespaces/NavigationProviderBase.cs

```diff
@@ -492,7 +493,7 @@ protected string MakePath(string parent, string child, bool childIsLeaf)
         /// for the provider namespace. For example, the file system provider should look
         /// for the last "\" and return everything to the left of the "\".
         /// </remarks>
-        protected virtual string GetParentPath(string path, string root)
+        protected virtual string GetParentPath(string path,int depth, string root)
```

Comment:
Formatting.
And maybe add new overload too.

Reply:
Sorry, new to C#.
So I need to create a new overload for ParseParent() has to be created and need to format the invocation lines ?

Reply:
@kvprasoon Please see docs https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/member-overloading
> DO make only the longest overload virtual (if extensibility is required). Shorter overloads should simply call through to a longer overload. 

So `GetParentPath(string path, string root)` should call `virtual GetParentPath(string path,int depth, string root)`